pkg/controller/controller_test.go

```diff
@@ -1819,17 +1866,19 @@ func TestProvisionFromSnapshot(t *testing.T) {
 		}
 
 		if tc.expectedPVSpec != nil {
-			if pv.Name != tc.expectedPVSpec.Name {
-				t.Errorf("test %q: expected PV name: %q, got: %q", k, tc.expectedPVSpec.Name, pv.Name)
-			}
+			if pv != nil {
+				if pv.Name != tc.expectedPVSpec.Name {
```

Comment:
Ideally, we should be comparing the whole PV object, not just the PVSource, and we should just do a deep equal on the entire object, not just individual fields.  That way, we won't miss validating new fields.

Reply:
ReclaimPolicy and AccessModes must to be compared？

Reply:
Actually is volume content supposed to be populated in PV object? That's what I had in mind.

If not then nevermind, this is more of a test cleanup issue

Reply:
I agree it that volume content should be populated in pv object, actually the volume content not to be populated, so I hadn't compare it

Reply:
@xing-yang @jingxu97 can you clarify if volumecontent is supposed to be populated in PV object, or only PVC object?

Reply:
@msau42 VolumeContentSource is not populated in PV object.  Only PVC has a DataSource.  We should probably take another look of this during API review.

Reply:
Thanks for clarifying. I think this PR is ready to go then